books/kestrel/number-theory/tonelli-shanks-proof.lisp

```diff
@@ -1194,7 +1194,7 @@
                 (equal (tonelli-shanks-sqrt-aux n p z) y))
            (if (has-square-root? n p)
 	       (equal (mod (* y y) p) n)
-	     0))
+	     (equal y 0)))
```

Comment:
Changed the theorem to prove that the algorithm returns 0 if n does not have a square root